modules/Bio/EnsEMBL/Hive/HivePipeline.pm

```diff
@@ -544,151 +544,206 @@ sub print_diagram {
 sub apply_tweaks {
     my $self    = shift @_;
     my $tweaks  = shift @_;
-
+    my @response;
+    my $responseStructure;
     my $need_write = 0;
-
+    $responseStructure->{Tweaks} = [];
     foreach my $tweak (@$tweaks) {
-        print "\nTweak.Request\t$tweak\n";
+        push @response, "Tweak.Request  \t$tweak\n";
 
         if($tweak=~/^pipeline\.param\[(\w+)\](\?|#|=(.+))$/) {
             my ($param_name, $operator, $new_value_str) = ($1, $2, $3);
-
+            my $tweakStructure;
+            $tweakStructure->{Object}->{Type} = 'Pipeline';
+            $tweakStructure->{Object}->{Id} = undef;
+            $tweakStructure->{Object}->{Name} = undef;
+            $tweakStructure->{Return}->{Field} = $param_name;
             my $pwp_collection  = $self->collection_of( 'PipelineWideParameters' );
             my $hash_pair       = $pwp_collection->find_one_by('param_name', $param_name);
-
             if($operator eq '?') {
-                print "Tweak.Show    \tpipeline.param[$param_name] ::\t"
-                     . ($hash_pair ? $hash_pair->{'param_value'} : '(missing_value)') . "\n";
+                my $value = $hash_pair ? $hash_pair->{'param_value'} : undef;
+                $tweakStructure->{Action} = 'SHOW';
+                $tweakStructure->{Return}->{OldValue} = $value;
+                $tweakStructure->{Return}->{NewValue} = $value;
+                push @response, "Tweak.Show  \tpipeline.param[$param_name] ::\t" . $value ? $value : '--> (missing value)' . "\n";
```

Comment:
* Remove `-->`
* `$value` can be an empty string

so do `$value // '(missing value)'`

Reply:
I have just left the same msg as it was before, not to confuse those who already use it